python_modules/dagster/dagster/_core/instance/config.py

```diff
@@ -36,6 +36,31 @@ def is_dagster_home_set() -> bool:
     return bool(os.getenv("DAGSTER_HOME"))
 
 
+def dagster_instance_class_from_code_pointer(
+    module_name: str, class_name: str
+) -> Type["DagsterInstance"]:
+    try:
+        return cast(
+            Type["DagsterInstance"],
+            class_from_code_pointer(module_name, class_name),
+        )
+    except DagsterImportClassFromCodePointerError as e:
+        module_elems = module_name.split(".")
+        if "dagster_cloud" == module_elems[0]:
+            try:
+                return cast(
+                    Type["DagsterInstance"],
+                    dagster_instance_class_from_code_pointer(
```

Comment:
can this call class_from_code_pointer directly rather than calling itself recursively?

Reply:
good call